controllers/articlesController.js

```diff
@@ -27,4 +30,21 @@ const patchArticleById = (req, res, next) => {
       next(err);
     });
 };
-module.exports = { getArticleById, patchArticleById };
+const getArticles = (req, res, next) => {
+  const { topic } = req.query;
+
+  const promises = [selectArticles(topic)];
+  if (topic) {
+    promises.push(selectTopicByName(topic));
+  }
+
+  Promise.all(promises)
+    .then((promises) => {
```

Comment:
I would recommend doing some destructing here, I dont think that promises[0] is particularly descriptive as to what is happening here